src/locale.js

```diff
@@ -76,23 +76,40 @@ export default class Locale {
 		this.contentLanguageDirection = getLanguageDirection( this.contentLanguage );
 
 		/**
-		 * Translates the given string to the {@link #uiLanguage}. This method is also available in
+		 * Translates the given message to the {@link #uiLanguage}. This method is also available in
 		 * {@link module:core/editor/editor~Editor#t} and {@link module:ui/view~View#t}.
 		 *
-		 * The strings may contain placeholders (`%<index>`) for values which are passed as the second argument.
+		 * This method's context is statically bound to the `Locale` instance and **always should be called as a function**:
+		 *
+		 *		const t = locale.t;
+		 *		t( 'Label' );
+		 *
+		 * The message can be either a string or an object implementing the {@link module:translation-service~Message} interface.
+		 *
+		 * A Message may contain placeholders (`%<index>`) for values that are passed as the second argument.
 		 * `<index>` is the index in the `values` array.
 		 *
-		 *		editor.t( 'Created file "%0" in %1ms.', [ fileName, timeTaken ] );
+		 *		t( 'Created file "%0" in %1ms.', [ fileName, timeTaken ] );
 		 *
-		 * This method's context is statically bound to Locale instance,
-		 * so it can be called as a function:
+		 * A Message can provide a plural form using the `plural` property and a value - that should be always the first element
```

Comment:
```suggestion
		 * The message support plural forms. To specify a plural form, use `plural`. Single or plural form will be chosen depending on the first value from the passed `values`.
```

Reply:
How about that?

Reply:
(PS. Introduce the change in code, not by committing this suggestion, as the length of this line will probably be too big.)

Reply:
(And merge my proposal with the lines below, where you describe how it may work for different languages ;).)